### PR TITLE
[AutoMM] Fix object detection prediction on COCO json files without annotations.

### PR DESCRIPTION
**Pull request by FANGAreNotGnu**
Fix object detection prediction on COCO json files without annotations.
Also add a test case for this issue.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/utils/object_detection.py [hunk 1]
```diff
@@ -778,6 +786,11 @@ def from_coco(
             use_crowd=use_crowd,
         )
         if not rois:
+            # discard the rows without valid annotation ONLY when data has annotation
+            # add default placeholder to data without annotation for prediction
+            if not num_annotations:
+                d["image"].append(abs_path)
+                d["rois"].append([[-1, -1, -1, -1, 0]])  # TODO: maybe remove this placeholder
```
Comment (tonyhoo) [inline multimodal/src/autogluon/multimodal/utils/object_detection.py:793]: Can you give me some context on why this specific placeholder is required

File: multimodal/tests/unittests/others/test_object_detection.py [hunk 2]
```diff
@@ -148,6 +149,16 @@ def test_mmdet_object_detection_inference_df(checkpoint_name):
...
+
+    test_path_without_annotations = os.path.join(data_dir, "Annotations", "test_cocoformat_nolabel.json")
+
+    with open(test_path, "r") as f:
+        test_data = json.load(f)
+    test_data.pop("annotations")
+    test_data["images"] = test_data["images"][:100]
+    with open(test_path_without_annotations, "w+") as f_wo_ann:
+        json.dump(test_data, f_wo_ann)
+
```
Comment (zhiqiangdon) [inline multimodal/tests/unittests/others/test_object_detection.py:161]: Can we modify the predictions in tutorials to reflect this support?

File: multimodal/src/autogluon/multimodal/utils/object_detection.py (hunk 1 above)
Comment (FANGAreNotGnu) [reply multimodal/src/autogluon/multimodal/utils/object_detection.py:793]: "rois" is the required feature column during training and evaluation but not for prediction. The requirement is defined here regardless of if it's for training, evaluation, or prediction: https://github.com/FANGAreNotGnu/autogluon/blob/master/multimodal/src/autogluon/multimodal/data/preprocess_dataframe.py#L200-L213 . We have the default placeholder added also for dict/list inputs, e.g. https://github.com/FANGAreNotGnu/autogluon/blob/master/multimodal/src/autogluon/multimodal/utils/object_detection.py#L213-L274

File: multimodal/tests/unittests/others/test_object_detection.py (hunk 2 above)
Comment (FANGAreNotGnu) [reply multimodal/tests/unittests/others/test_object_detection.py:161]: Updated the doc a bit. I'll add a new tutorial on what kind of data we can take for prediction (inference).

**Review by tonyhoo — APPROVED** [review]

**Pull request merged**